lib/galaxy/datatypes/interval.py

```diff
@@ -1447,6 +1451,116 @@ def sniff(self, filename):
+                & ('pos' in columns or 'pos1' in columns or 'start' in columns or 'start1' in columns)
+                & ('chr2' in columns)
+                & ('pos2' in columns or 'start2' in columns))
+
+    def sniff_prefix(self, file_prefix):
+        return self.sniff(file_prefix.filename)
+
+    def set_meta(self, dataset, **kwd):
+        """ Creates the index for the pairix file. """
+        # We need to know which columns correspond to the chromosome name,
+        # the start coordinate, and if a stop coordinate is present.
+        # In order to produce a pairx index, we need to look at the
+        # required header line "#columns:". The sniffer should already
+        # have ensured that this line is present.
+        Tabular.set_meta(self, dataset, skip=0)
+        file_iterator = util.compression_utils.get_fileobj(dataset.file_name)
+        sep_character = None
+        comment_lines = 0
+        for line in file_iterator:
+            log.info(line)
+            if sep_character is not None:
+                break
+            elif len(line) > 0 and line[0] != '#':
+                fields = line.split()
+                if len(fields) > 1:
+                    sep_character = line.split(fields[1])[0].split(fields[0])[-1]
+            else:
+                comment_lines += 1
+                line = line.rstrip('\n\r')
+                if line.startswith('#columns: '):
+                    line = line.lower().replace('chrom', 'chr').replace('chromosome', 'chr').replace('stop', 'end')
+                    columns = line.split()[1:]
+        dataset.metadata.delimiter = sep_character
+        dataset.metadata.comment_lines = comment_lines
+        dataset.metadata.column_names = columns
+        dataset.metadata.columns = len(columns)
+        for i, name in enumerate(columns):
+            if name == 'chr' or name == 'chr1':
+                dataset.metadata.chromCol = i + 1
+            elif name == 'pos' or name == 'pos1':
+                dataset.metadata.startCol = i + 1
+                dataset.metadata.endCol = i + 1
+            elif name == 'start' or name == 'start1':
+                dataset.metadata.startCol = i + 1
+            elif name == 'end' or name == 'end1':
+                dataset.metadata.endCol = i + 1
+            elif name == 'chr2':
+                dataset.metadata.chrom2Col = i + 1
+            elif name == 'pos2':
+                dataset.metadata.start2Col = i + 1
+                dataset.metadata.end2Col = i + 1
+            elif name == 'start2':
+                dataset.metadata.start2Col = i + 1
+            elif name == 'end2':
+                dataset.metadata.end2Col = i + 1
+        index_file = dataset.metadata.pairix_index
+        if not index_file:
+            index_file = dataset.metadata.spec['pairix_index'].param.new_file(dataset=dataset)
+        try:
+            pypairix.build_index(dataset.file_name, sc=dataset.metadata.chromCol, bc=dataset.metadata.startCol,
```

Comment:
Is it absolutely necessary to add pypairix ? This seems like quite a heavy-weight dependency, it would be great if we could avoid this. If the indexing isn't very heavy this could be done in the tool itself for instance.

Reply:
You mean add the pypairix dependency *to the tool* instead of galaxy?

Reply:
Yes, and move the index generation to the tool as well.

Reply:
Yes, makes sense. @msauria I'll be happy to help if needed.